src/modals/DetailsModal/DetailsModal.js

```diff
@@ -18,29 +18,35 @@ export default function DetailsModal({
 
   useEffect(() => {
     if (node.data.wikipediaSlug)
-      getData(
-        `https://${currentLang.code}.wikipedia.org/api/rest_v1/page/summary/${node.data.wikipediaSlug}`
-      ).then(({ extract, thumbnail }) => {
-        if (extract) setWikipediaExtract(extract);
-        if (thumbnail && !images.length) {
-          setImages({
-            url: thumbnail.source,
-            alt: `${node.data.label}'s Wikipedia image`,
-          });
+      getWikipediaArticle(node.data.wikipediaSlug, currentLang.code).then(
+        ({ extract, thumbnail }) => {
+          if (extract) setWikipediaExtract(extract);
+          if (thumbnail && !images.length) {
+            setImages({
+              url: thumbnail.source,
+              alt: `${node.data.label}'s Wikipedia image`,
+            });
+          }
         }
-      });
+      );
+  }, [
+    currentLang.code,
+    images.length,
+    node.data.wikipediaSlug,
+    node.data.label,
+  ]);
 
+  useEffect(() => {
     if (node.data.birthPlaceId || node.data.deathPlaceId) {
-      getLabels([node.data.birthPlaceId, node.data.deathPlaceId], currentLang.code).then(entities => {
-        if (node.data.birthPlaceId) {
-          setBirthPlace(entities[node.data.birthPlaceId].labels[currentLang.code].value);
-        }
-        if (node.data.deathPlaceId) {
-          setDeathPlace(entities[node.data.deathPlaceId].labels[currentLang.code].value);
-        }
+      getItemsLabels(
+        [node.data.birthPlaceId, node.data.deathPlaceId],
+        currentLang.code
+      ).then(([birthPlaceLabel, deathPlaceLabel]) => {
+        setBirthPlace(birthPlaceLabel);
+        setDeathPlace(deathPlaceLabel);
```

Comment:
Did you test it for the case, in which only the birthplace is set?

Reply:
yes it will set `undefined` and the jsx will not display

Reply:
Ok then